src/grpc/lightning.ts

```diff
@@ -364,10 +365,35 @@ export function keysend(
             } else {
               if (state === 'IN_FLIGHT') {
                 // do nothing
-              } else if (state === 'FAILED_NO_ROUTE') {
-                reject(payment.failure_reason || payment)
-              } else if (state === 'FAILED') {
-                reject(payment.failure_reason || payment)
+              } else if (state === 'FAILED_NO_ROUTE' || state === 'FAILED') {
+                delete options.payment_hash
+                delete options.dest_custom_records[`${LND_KEYSEND_KEY}`]
+                delete options.dest_features
```

Comment:
ok so if you keep these values in it'll lead to the error `destination doesn't support AMP payments`?

If so that error message is very misleading because the destination does support AMP payments just the construction of the amp payment is wrong

Reply:
i think the destination gives some error 'not supported' and the sender assumes its amp thats not supported

Reply:
https://github.com/stakwork/sphinx-relay/pull/362#issuecomment-1102675439

i basically removed everything of KeysendOptions not in this amp test and it worked

Reply:
ok lemme open an issue for LND aswell since that really led me down the wrong path

Reply:
```
lnrpc/routerrpc/router_backend.go
                 821-				// First make sure the destination supports AMP.
                 822-				if !features.HasFeature(lnwire.AMPOptional) {
                 823-					return nil, fmt.Errorf("destination doesn't " +
                 824:						"support AMP payments")
                 825-				}
```
apparently it looks at the feature bits

Reply:
i think some number had to be added to `options.dest_features` then to get it working with that

Reply:
yea I saw that it was doing that but arent the feature bits setup during the startup of the lnd node?

Reply:
```
lnwire/features.go
                 134-		// sender-generated preimages according to BOLT XX.
                 135-		AMPRequired FeatureBit = 30
                 136-
                 137:		// AMPOptional is an optional feature bit that signals that the receiver
                 138-		// of a payment supports accepts spontaneous payments, i.e.
                 139-		// sender-generated preimages according to BOLT XX.
                 140:		AMPOptional FeatureBit = 31
```

one of these (30, 31)

Reply:
and if you add 30 and 31 to dest_features? does it even work?

Reply:
not sure lets try it lol